language-design.md

```diff
@@ -49,6 +68,11 @@ In what way is an API a language?
 
 **Response**
 
+According to Wikipedia, by definition an API, or application programming interface, is “a set of routines, protocols, and tools for building software and applications” [[Wikipedia]]( https://en.wikipedia.org/wiki/Application_programming_interface). When considering this definition along the comments of Joshua Bloch, it becomes clear there isn’t a big distinction between a good API and an internal DSL. In fact, these terms are even more blurred when Bloch describes an API as, “a little language, and people must learn to read and write it” [Bloch, 2006, 506]. Just as an internal DSL is a language embedded within a host language, APIs are exposed “intermodular boundaries” [Bloch, 2006, 506] written in another language for module communication. Therefore, an API, like an internal DSL, is written in a language and contains an abstracted subset of the “host language” to be used alongside other “host language” code (I quoted “host language” since this is a term usually used for the language an internal DSL is implemented in, not the language an API was written in).
+
+Furthermore, unlike a general purpose programming language, an API has a very small focus. Convoluting an API with general functionality is almost always a detriment. It is for this reason, Joshua’s advice often follows the KISS paradigm (Keep It Simple Stupid). While Joshua’s advice of “Keep APIs free of implementation details,” “When in doubt, leave it out,” and “Minimize accessibility; when in doubt, make it private” [Bloch, 2006, 506], is all given in the context of good API design, we can also validly apply these recommendations to a focused DSL. For instance, all of these points can also be made in reference to the design philosophy of ContextFree. Specifically, although ContextFree is a powerful tool for designing digital art, it is limited to just this task. That is, ContextFree intentionally does not enable users to access low level features and implementation details or complete non-digital art tasks.
```

Comment:
This comparison with ContextFree is a good piece of evidence for what you're saying here. That said, even though I agree that an API is more comparable to a DSL, I think most of this advice really applies to general purpose PLs too. It's a sort of different application of the paradigm, where the design space for a DSL is more restricted.